pkg/fswatcher/fs_watcher.go

```diff
@@ -16,40 +16,49 @@ package fswatcher
 
 import "github.com/fsnotify/fsnotify"
 
-// Watcher watches for Events and Errors once a resource is added to the watch list.
-// Primarily used for mocking the fsnotify lib.
 type Watcher interface {
-	Add(name string) error
+	WatchFiles(paths []string, onChange func(), onRemove func()) error
 	Close() error
-	Events() chan fsnotify.Event
-	Errors() chan error
 }
 
 // fsnotifyWatcherWrapper wraps the fsnotify.Watcher and implements Watcher.
 type fsnotifyWatcherWrapper struct {
 	fsnotifyWatcher *fsnotify.Watcher
 }
 
-// Add adds the filename to watch.
-func (f *fsnotifyWatcherWrapper) Add(name string) error {
-	return f.fsnotifyWatcher.Add(name)
+func (f *fsnotifyWatcherWrapper) WatchFiles(paths []string, onChange func(), onRemove func()) error {
+	go func() error {
```

Comment:
Yeah, that was stated in your instruction. But I just wonder if all that complexity is specific to `cert_watcher` or should be applied to all that use the API? For example, does `static_handler` really need that complexity?

Reply:
What the recent changes in `cert_watcher` revealed was that the original naive implementation of the watcher did not work in some environments, depending on how rotation of the certificates was done. And since we have to have this more complex implementation of the watcher done for certificates anyway, it doesn't make sense to have another one, more naive, for other use cases, that may eventually also run into the same issue.